src/libraries/Common/src/System/Net/SocketProtocolSupportPal.cs

```diff
@@ -12,7 +12,7 @@ internal static partial class SocketProtocolSupportPal
 
         public static bool OSSupportsIPv6 { get; } = IsSupported(AddressFamily.InterNetworkV6) && !IsIPv6Disabled();
         public static bool OSSupportsIPv4 { get; } = IsSupported(AddressFamily.InterNetwork);
-        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix);
+        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix) && !OperatingSystem.IsTvOS() && (!OperatingSystem.IsIOS() || OperatingSystem.IsMacCatalyst());
```

Comment:
nit: move the OS checks to the IsSupported method in SocketProtocolSupportPal.Unix.cs